core/core_bind.cpp

```diff
@@ -1224,6 +1224,73 @@ void Mutex::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("unlock"), &Mutex::unlock);
 }
 
+////// SharedMutex //////
+
+void SharedMutex::lock_exclusive() {
+	exclusive_mutex.lock();
```

Comment:
Calling lock shared mutex without unlocking and then requesting exclusive access results in a deadlocked state.

Reply:
Documented.